AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCosmosDb.m

```diff
@@ -0,0 +1,136 @@
+  return (NSString *)[string stringByAddingPercentEncodingWithAllowedCharacters:allowedCharacters];
+}
+
++ (NSString *)rfc1123String:(NSDate *)date {
+  static NSDateFormatter *df = nil;
+  if (!df) {
+    static dispatch_once_t oncePredicate;
+    dispatch_once(&oncePredicate, ^{
+      df = [[NSDateFormatter alloc] init];
+      df.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US"];
+      df.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"GMT"];
+      df.dateFormat = @"EEE',' dd MMM yyyy HH':'mm':'ss 'GMT'";
+    });
+  }
+  return [df stringFromDate:date];
+}
+
++ (NSDictionary *)defaultHeaderWithPartition:(NSString *)partition dbToken:(NSString *)dbToken {
+  return @{
+    kMSHeaderDocumentDbPartitionKey : [NSString stringWithFormat:kMSHeaderDocumentDbPartitionKeyFormat, partition],
+    kMSHeaderMsVesion : kMSHeaderMsVesionValue,
+    kMSHeaderMsDate : [MSCosmosDb rfc1123String:[NSDate date]],
+    kMSHeaderContentTypeKey : kMSAppCenterContentType,
+    kMSHeaderAuthorization : [MSCosmosDb encodeUrl:dbToken]
+  };
+}
+
++ (NSString *)documentDbEndpointWithDbAccount:(NSString *)dbAccount documentResourceId:(NSString *)documentResourceId {
+  NSString *documentEndpoint = [NSString stringWithFormat:kMSDocumentDbEndpoint, dbAccount];
+  return [NSString stringWithFormat:@"%@/%@", documentEndpoint, documentResourceId];
+}
+
++ (NSString *)documentBaseUrlWithDatabaseName:(NSString *)databaseName
+                               collectionName:(NSString *)collectionName
+                                   documentId:(NSString *)documentId {
+  NSString *dbUrlSuffix = [NSString stringWithFormat:kMSDocumentDbDatabaseUrlSuffix, databaseName];
+  NSString *dbCollectionUrlSuffix = [NSString stringWithFormat:kMSDocumentDbCollectionUrlSuffix, collectionName];
+  NSString *dbDocumentId = documentId ? [NSString stringWithFormat:@"/%@", documentId] : @"";
+  return [NSString stringWithFormat:@"%@/%@/%@%@", dbUrlSuffix, dbCollectionUrlSuffix, kMSDocumentDbDocumentUrlPrefix, dbDocumentId];
+}
+
++ (NSString *)documentUrlWithTokenResult:(MSTokenResult *)tokenResult documentId:(NSString *)documentId {
+  NSString *documentResourceIdPrefix = [MSCosmosDb documentBaseUrlWithDatabaseName:tokenResult.dbName
+                                                                    collectionName:tokenResult.dbCollectionName
+                                                                        documentId:documentId];
+  return [MSCosmosDb documentDbEndpointWithDbAccount:tokenResult.dbAccount documentResourceId:documentResourceIdPrefix];
+}
+
++ (void)cosmosDbAsync:(MSCosmosDbIngestion *)httpIngestion
+          tokenResult:(MSTokenResult *)tokenResult
+           documentId:(NSString *)documentId
+             httpVerb:(NSString *)httpVerb
+                 body:(NSString *)body
+    completionHandler:(MSCosmosDbCompletionHandler)completion {
+
+  // Configure http client.
```

Comment:
All 3 comments in this method feel unnecessary to me. Consider removing them as they are stating the obvious to me. I'm not too pushy here because maybe your intention is to structure the code and I see the value of that.

Reply:
it's intended to structure the code and increase readability